### PR TITLE
also instantiate the slave and coreos-assembler in the template

### PR DESCRIPTION
**Pull request by jlebon**
This makes the set up instructions much easier and allows us finer
grained control over the resources in the future if we'd like. One
notable change here is `lookupPolicy.local: true`, which allows us to
directly reference the imagestream from the pod. This conveniently also
works around the registry DNS issues mentioned in #3.

### CODE REVIEW

**Comment by jlebon** [comment]
Rebased!

**Comment by jlebon** [comment]
This one should unlock a lot of the remaining PRs.

File: HACKING.md [hunk 1]
```diff
@@ -31,7 +31,10 @@ $ oc new-app --file=manifests/bc.yaml \
 Right now this template creates two buildconfigs:
```
Comment (dustymabe) [inline HACKING.md]: there are 5 things in the list below. maybe update this comment

File: HACKING.md [hunk 2]
```diff
@@ -47,34 +50,13 @@ It is a good idea to
 and 
 [set the kubernetes plugin to use DNS for service names](TROUBLESHOOTING.md#issue-for-jenkins-dns-names).
```
Comment (dustymabe) [inline HACKING.md]: are the statements above needed any longer?

File: manifests/bc.yaml [hunk 3]
```diff
@@ -73,6 +73,65 @@ objects:
...
+  # keep a local copy of the Jenkins slave image so we're not constantly pulling
+  # it each time from docker.io
+  - apiVersion: v1
+    kind: ImageStream
+    metadata:
+      name: jenkins-slave-base-centos7
+      namespace: fedora-coreos
+    spec:
+      lookupPolicy:
+        # this allows e.g. the pipeline to directly reference the imagestream
```
Comment (dustymabe) [inline manifests/bc.yaml]: nice - didn't know about this feature

**Comment by dustymabe** [comment]
> This conveniently also
> works around the registry DNS issues mentioned in #3.

nice. does that mean we could close #3 after this is merged?

File: HACKING.md [hunk 4]
```diff
@@ -31,7 +31,10 @@ $ oc new-app --file=manifests/bc.yaml \
 Right now this template creates two buildconfigs:
 
 1. The Jenkins master imagestream
-2. The Jenkins pipeline build
+1. The Jenkins slave imagestream
+1. The coreos-assembler imagestream
+2. The PVC
+3. The Jenkins pipeline build
```
[the remark this answers is not in the document]
Comment (dustymabe) [reply HACKING.md]: you did this already with 1/2/3/4/5

File: HACKING.md (hunk 2 above)
Comment (jlebon) [reply HACKING.md]: I deleted the one about updating the plugins but kept the DNS one. I haven't hit it myself, though I don't think switching to s2i would somehow fix that. Eventually, we could likely tweak the s2i to include this as part of the built master image.

**Comment by jlebon** [comment]
> nice. does that mean we could close #3 after this is merged?

#21 will close that one. :)

**Pull request merged**